eclipse-che/che-theia
# chore: update local git config when user changes Che Theia git settings

### What does this PR do?
* When user gitconfig `home/theia/.gitconfig` is updated with a new value, local gitconfig `<project directory>/.git/config` is updated with the value automatically.
* When an existing value is updated in user gitconfig:
  * If the old value is equal with one in local gitconfig, it is updated in the local gitconfig.
  * If the old value is NOT equal with one in local gitconfig, it is NOT updated in the local gitconfig.
* When a value is deleted in user gitconfig:
  * If the value is equal with one in local gitconfig, it is deleted in the local gitconfig.
  * If the value is NOT equal with one in local gitconfig, it is NOT deleted in the local gitconfig.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/21115

### How to test this PR?

run factory: 
```
https://github.com/che-samples/bash/tree/devfilev2?che-editor=https://gist.githubusercontent.com/vinokurig/8240d845067c89e295e2cb4dc2c0070b/raw/ad3452ee49b0eeddfaf086f55557d46206073fea/che-theia.yaml
```
1. Set some `gitconfig` settings e.g. `user.name` and `user.email`.
2. Restart workspace.

See: The `gitconfig` settings are present.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)



### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
